### ISSUE TITLE
Can we use the drop-in component without bringing in all the dependencies like the WeChat SDK which we may not be interested in?

### ISSUE THREAD

**Issue by StylianosGakis** [Adyen/adyen-android#1172]
**Is your feature request related to a problem? Please describe.**
When building the project, there's always warnings from the wechat sdk, speficially `Expected stack map table for method with non-linear control flow. In later version of R8, the method may be assumed not reachable.`
And
```
AGPBI: {"kind":"warning","text":"Expected stack map table for method with non-linear control flow. In later version of R8, the method may be assumed not reachable.","sources":[{"file":"/Users/___/.gradle/caches/transforms-3/f3d6362884a684501f1bbd8a06dce763/transformed/wechat-sdk-android-without-mta-6.6.4-runtime.jar"}],"tool":"R8"}
```
With gradle 8.x, and strict R8 mode by default we even can't build the project without adding the proguard rules that the wechat sdk requires.

Specifically with the gradle bump, I get
```
Missing class org.slf4j.impl.StaticLoggerBinder (referenced from: void org.slf4j.LoggerFactory.bind() and 3 other contexts)
```
from the `wechat-sdk-android-without-mta-6.6.4-runtime.jar` jar.
And by running `./gradlew :app:dependencies` I see that the only way we're bringing in the wechat dependency from `com.adyen.checkout:3ds2:4.10.0` which comes from `com.adyen.checkout:drop-in:4.10.0`

**Describe the solution you'd like**
Optimally, bringing in only parts of the SDK that I am interested in would be the best solution.

**Describe alternatives you've considered**
I've considered doing something like
```
implementation(adyen) {
  exclude(wechat-sdk)
}
```
To remove the dependency myself, but I am afraid this might have bad side effects on the Adyen SDK itself, and wouldn't feel safe to do so.

**Additional context**
If there's some way for me to already do this that I am not aware of, please do let me know.
For now, I've added `-dontwarn org.slf4j.impl.StaticLoggerBinder` to my proguard, but that only feels like a workaround tbh, and I'm not that confident that it's what I want in the first place.

**Comment by OscarSpruit** [comment]
Hi @StylianosGakis, thank you for brining this to our attention! I agree with you that you should be able to exclude dependencies that you don't want. However, at the moment when excluding, for example WeChat, drop-in will crash at some point during runtime. Our vision for drop-in is that it works out of the box for every payment method, so the final solution for this problem will probably be making it safe for you to exclude unwanted dependencies. We are going to investigate the possibilities and we will get back to you once we have more.

**Comment by StylianosGakis** [comment]
Hello Oscar, thanks for getting back at me this fast!

Yes, I understand that philosophy, and it does make sense. The idea of possibly manually excluding things I know I do not need are also something that would work for me, for sure. It could also help you on your end with less maintenance needed for such issues.

For example now that a lot of folks are going to start using gradle 8 and AGP 8 too, where R8 is on full mode by default, basically all of them are probably going to be facing the exact same issue as I was, so you probably have to assist a lot of folks due to a problem that an external SDK is introducing which I assume a big chunk of your users especially from world regions that don't use WeChat a lot don't need anyway.

Hope something comes out of this, please do keep us posted here if anything changes or you want feedback on some options you may be considering 😊

**Comment by StylianosGakis** [comment]
Thanks for doing this, a really nice step forward.
I am trying 4.12.0 now, and even when I try this approach of removing the wechat and the 3ds2 dependency, when I try to build a minified build, I still do get a stacktrace like this in the `:app:minifyReleaseWithR8` gradle task. Have you checked how this change works with minified r8 builds? Am I doing something wrong here? 
For reference, I did also try to add or remove the line `-dontwarn org.slf4j.impl.StaticLoggerBinder` from my proguard rules, and all it does is that it stops warning about that one in particular when this rule is there.

```
Missing class com.adyen.checkout.adyen3ds2.Adyen3DS2Component (referenced from: com.adyen.checkout.components.ActionComponentProvider com.adyen.checkout.dropin.ComponentParsingProviderKt.getActionProviderFor(com.adyen.checkout.components.model.payments.response.Action))
Missing class com.adyen.checkout.adyen3ds2.Adyen3DS2Configuration$Builder (referenced from: com.adyen.checkout.components.base.BaseActionComponent com.adyen.checkout.dropin.ComponentParsingProviderKt.getActionComponentFor(androidx.fragment.app.FragmentActivity, com.adyen.checkout.components.ActionComponentProvider, com.adyen.checkout.dropin.DropInConfiguration) and 1 other context)
Missing class com.adyen.checkout.adyen3ds2.Adyen3DS2Configuration (referenced from: com.adyen.checkout.components.base.BaseActionComponent com.adyen.checkout.dropin.ComponentParsingProviderKt.getActionComponentFor(androidx.fragment.app.FragmentActivity, com.adyen.checkout.components.ActionComponentProvider, com.adyen.checkout.dropin.DropInConfiguration) and 1 other context)
Missing class com.adyen.checkout.wechatpay.WeChatPayActionComponent (referenced from: com.adyen.checkout.components.ActionComponentProvider com.adyen.checkout.dropin.ComponentParsingProviderKt.getActionProviderFor(com.adyen.checkout.components.model.payments.response.Action))
Missing class com.adyen.checkout.wechatpay.WeChatPayActionConfiguration$Builder (referenced from: com.adyen.checkout.components.base.BaseActionComponent com.adyen.checkout.dropin.ComponentParsingProviderKt.getActionComponentFor(androidx.fragment.app.FragmentActivity, com.adyen.checkout.components.ActionComponentProvider, com.adyen.checkout.dropin.DropInConfiguration) and 1 other context)
Missing class com.adyen.checkout.wechatpay.WeChatPayActionConfiguration (referenced from: com.adyen.checkout.components.base.BaseActionComponent com.adyen.checkout.dropin.ComponentParsingProviderKt.getActionComponentFor(androidx.fragment.app.FragmentActivity, com.adyen.checkout.components.ActionComponentProvider, com.adyen.checkout.dropin.DropInConfiguration) and 1 other context)
Missing class com.adyen.checkout.wechatpay.WeChatPayProvider (referenced from: com.adyen.checkout.components.PaymentMethodAvailabilityCheck com.adyen.checkout.dropin.ComponentParsingProviderKt.getPaymentMethodAvailabilityCheck(java.lang.String))
Missing class com.adyen.checkout.wechatpay.WeChatPayUtils (referenced from: void com.adyen.checkout.dropin.ui.DropInActivity.handleIntent(android.content.Intent))
Missing class com.adyen.threeds2.ThreeDS2Service (referenced from: com.adyen.checkout.components.GenericComponentState com.adyen.checkout.bcmc.BcmcComponent.createComponentState() and 1 other context)
```

**Comment by jreij** [comment]
Hi @StylianosGakis, we just released [5.0.0-alpha02](https://github.com/Adyen/adyen-android/releases/tag/5.0.0-alpha02) with the ability to exclude action modules that contain 3rd party dependencies, check out the release notes for more info!
